typify/tests/schemas.rs

```diff
@@ -64,7 +64,8 @@ fn validate_schema(path: std::path::PathBuf) -> Result<(), Box<dyn Error>> {
                 "std",
                 typify::CrateVers::Version("1.0.0".parse().unwrap()),
                 None,
-            ),
+            )
+            .with_struct_builder(true),
```

Comment:
See, see my comment https://github.com/oxidecomputer/typify/pull/705#issuecomment-2512335987 for a detailed explanation.  There was one place left in the codegen where `String` was used, and all of the tests passed because that one place was in the codegen for optional struct builders.  So I enabled that for all of the schema tests, the tests started to fail, then I fixed that one remaining use of a simple `String` type and verified that the tests passed.  I left this in place because it seems like these tests are meant to exercise the code generation in its entirety.